docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -241,236 +287,447 @@ This generates the ABI code and moves it to `./contract-examples/contracts` as `
-
-We will be getting and setting our greeting with `sayHello()` and `setGreeting()` in one slot
-respectively so we can define the gas costs as follows.
+##### Verify()
 
-```go
-	SayHelloGasCost    uint64 = readGasCostPerSlot
-	SetGreetingGasGost uint64 = writeGasCostPerSlot
-```
-
-### Step 3: Add Custom Code
+`Verify()` is called on startup and an error is treated as fatal. Generated code contains a call
+to `AllowListConfig.Verify()` to verify the `AllowListConfig`. You can leave that as is and start
+adding your own custom verify code after that.
 
-Search (`CTRL F`) throughout the file with `CUSTOM CODE STARTS HERE` to find the areas in the
-precompile that we need to modify. We should start with the reference imports code block.
+We can leave this function as is right now because there is no invalid custom configuration for the `Config`.
 
-![Custom code starts here](./../../static/img/2022-09-01-22-48-26.png)
+```go
+// Verify tries to verify Config and returns an error accordingly.
+func (c *Config) Verify() error {
+	// Verify AllowList first
+	if err := c.AllowListConfig.Verify(); err != nil {
+		return err
+	}
 
-We can remove all of these reference imports and the associated imports. We only include these in
-our template so that the file compiles on generation.
+	// CUSTOM CODE STARTS HERE
+	// Add your own custom verify code for Config here
+	// and return an error accordingly
+	return nil
+}
+```
 
-#### Step 3.1: Modify Equal()
+##### Equal()
 
-Next we see this in `Equal()`. The below code snippet can be copied and pasted to overwrite the
-default `Equal()` code.
+Next we see is `Equal()`. This function determines if two precompile configs are equal. This is used
+to determine if the precompile needs to be upgraded. There is some default code that is generated for
+checking `Upgrade` and `AllowListConfig` equality.
 
 <!-- markdownlint-disable MD013 -->
 
 ```go
-// Equal returns true if [s] is a [*HelloWorldConfig] and it has been configured identical to [c].
-func (c *HelloWorldConfig) Equal(s StatefulPrecompileConfig) bool {
+// Equal returns true if [s] is a [*Config] and it has been configured identical to [c].
+func (c *Config) Equal(s precompileconfig.Config) bool {
 	// typecast before comparison
-	other, ok := (s).(*HelloWorldConfig)
+	other, ok := (s).(*Config)
 	if !ok {
 		return false
 	}
 	// CUSTOM CODE STARTS HERE
-	// modify this boolean accordingly with your custom HelloWorldConfig, to check if [other] and the current [c] are equal
-	// if HelloWorldConfig contains only UpgradeableConfig  you can skip modifying it.
-	equals := c.UpgradeableConfig.Equal(&other.UpgradeableConfig)
+	// modify this boolean accordingly with your custom Config, to check if [other] and the current [c] are equal
+	// if Config contains only Upgrade  and AllowListConfig  you can skip modifying it.
+	equals := c.Upgrade.Equal(&other.Upgrade) && c.AllowListConfig.Equal(&other.AllowListConfig)
```

Comment:
nit: extra space

Reply:
ah that probably because of the ws in template